src/mmds/src/data_store.rs

```diff
@@ -58,6 +62,12 @@ impl Mmds {
         }
     }
 
+    pub fn set_data_store_limit(&mut self, maybe_data_store_limit: Option<usize>) {
```

Comment:
This should not take an `Option<usize>` but an `usize`. When someone is calling a set, they should have already validated that they indeed have a limit they want to enforce.

The check here should be done by the caller of this function.